node/sc/vt_recovery.go

```diff
@@ -271,13 +271,20 @@ pendingTxLoop:
 			endBlkNum = curBlkNum
 		}
 		opts := &bind.FilterOpts{Start: startBlkNum, End: &endBlkNum}
-		it, err := bi.bridge.FilterRequestValueTransfer(opts, nil, nil, nil)
+		it, err := from.bridge.FilterRequestValueTransfer(opts, nil, nil, nil)
 		if err != nil {
 			return nil, err
 		}
 		for it.Next() {
 			logger.Trace("pending nonce in the event", "requestNonce", it.Event.RequestNonce)
 			if it.Event.RequestNonce >= hint.handleNonce {
+				// Check if the event is already handled in target bridge contract
+				blk, err := to.bridge.HandleNoncesToBlockNums(nil, it.Event.RequestNonce)
+				if err == nil && blk > 0 {
```

Comment:
Typically, recovery will operate in an abnormal situation and will have to be treated as a batch, but looking up a contact at each event may be a performance disadvantage.